seacrowd/sea_datasets/cebuaner/cebuaner.py

```diff
@@ -0,0 +1,151 @@
+    """CebuaNER dataset from https://github.com/mebzmoren/CebuaNER"""
+
+    SOURCE_VERSION = datasets.Version(_SOURCE_VERSION)
+    SEACROWD_VERSION = datasets.Version(_SEACROWD_VERSION)
+
+    SEACROWD_SCHEMA_NAME = "seq_label"
+    LABEL_CLASSES = [
+        "O",
+        "B-PER",
+        "I-PER",
+        "B-ORG",
+        "I-ORG",
+        "B-LOC",
+        "I-LOC",
+        "B-OTHER",
+        "I-OTHER",
+    ]
+
+    BUILDER_CONFIGS = [
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_source",
+            version=SOURCE_VERSION,
+            description=f"{_DATASETNAME} source schema",
+            schema="source",
+            subset_id=_DATASETNAME,
+        ),
+        SEACrowdConfig(
+            name=f"{_DATASETNAME}_seacrowd_{SEACROWD_SCHEMA_NAME}",
+            version=SEACROWD_VERSION,
+            description=f"{_DATASETNAME} SEACrowd schema",
+            schema=f"seacrowd_{SEACROWD_SCHEMA_NAME}",
+            subset_id=_DATASETNAME,
+        ),
+    ]
+
+    DEFAULT_CONFIG_NAME = f"{_DATASETNAME}_source"
+
+    def _info(self) -> datasets.DatasetInfo:
+        if self.config.schema == "source":
+            features = datasets.Features(
+                {
+                    "id": datasets.Value("string"),
+                    "tokens": datasets.Sequence(datasets.Value("string")),
+                    "ner_tags": datasets.Sequence(datasets.features.ClassLabel(names=self.LABEL_CLASSES)),
+                }
+            )
+        elif self.config.schema == f"seacrowd_{self.SEACROWD_SCHEMA_NAME}":
+            features = schemas.seq_label_features(self.LABEL_CLASSES)
+
+        return datasets.DatasetInfo(
+            description=_DESCRIPTION,
+            features=features,
+            homepage=_HOMEPAGE,
+            license=_LICENSE,
+            citation=_CITATION,
+        )
+
+    def _split_generators(self, dl_manager: DownloadManager) -> List[datasets.SplitGenerator]:
+        data_file = Path(dl_manager.download_and_extract(_URL))
+        return [
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": data_file, "split": "train"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.VALIDATION,
+                gen_kwargs={"filepath": data_file, "split": "dev"},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": data_file, "split": "test"},
+            ),
+        ]
+
+    def _generate_examples(self, filepath: Path, split: str) -> Tuple[int, Dict]:
+        label_key = "ner_tags" if self.config.schema == "source" else "labels"
+        examples: Iterable[Dict[str, List[str]]] = []
+        with open(filepath, encoding="utf-8") as f:
+            tokens = []
+            ner_tags = []
+            for line in f:
+                if line.startswith("-DOCSTART-") or line == "" or line == "\n":
```

Comment:
@ljvmiranda921 
this is related to the data ```_URL```, defined. I think you made a good check on their datasource for this dataloader, however do you mind to add some comments to this line, to give clarity on the rationale to other devs & contributors? similar to what you did in data-splitting on train-val-test, which is a great comment imo!

Reply:
Updated, let me know what you think! https://github.com/SEACrowd/seacrowd-datahub/pull/27/commits/b146acc9e928ff4bfb56136e7a365cf2895ad798

Reply:
I found on [this file](https://github.com/mebzmoren/CebuaNER/blob/main/notebooks/Named-Entity-Recognition-with-Conditional-Random-Fields.ipynb) that in 3rd Code Block (titled Converting to list of list per article) they're using only `''` as article delimiter. I think you can either follow it or stick to the current implementation while adding this in your comment (I'm preferring the 2nd one tho)

Reply:
Updated! https://github.com/SEACrowd/seacrowd-datahub/pull/27/commits/e18deefe25b45aa810126e7936a7b38d342e35eb I went with the second option but referenced the notebook.